### DISCUSSION TITLE
Access to an URL with datas in method POST doesn't work

### DISCUSSION THREAD

**Discussion by laurent0502** [MicrosoftEdge/WebView2Feedback#3937] [Q&A]
Hello,

I've a problem with WebView2. To summarize, I use webview2 in my C++ application. In the documentation of ICoreWebView2, I haven't seen a function "NavigateWithPostData" or something like that. 

So, I tried with the function "Navigate" and then, in my "add_NavigationCompleted" listener, I tried : 

`m_controlsWebView->PostWebMessageAsJson(utils::stringToWstring(EdgeBrowser::_post).c_str());`

where EdgeBrowser::_post is : 

```
file={"features":[{"geometry":{"coordinates":[39.000000,7.000000],"type":"Point"},
"properties":{"Code postal":"2","Commune":"1","Coordonnee (X)":"39.0000000","Coordonnee (Y)":"7.0000000","Numero de maison":"7","Rue":"40"},"type":"Feature"},{"geometry":{"coordinates":[41.000000,7.000000],"type":"Point"},
"properties":{"Code postal":"4","Commune":"3","Coordonnee (X)":"41.0000000","Coordonnee (Y)":"7.0000000","Numero de maison":"7","Rue":"42"},"type":"Feature"},{"geometry":{"coordinates":[43.000000,7.000000],"type":"Point"},
"properties":{"Code postal":"6","Commune":"5","Coordonnee (X)":"43.0000000","Coordonnee (Y)":"7.0000000","Numero de maison":"7","Rue":"44"},"type":"Feature"},{"geometry":{"coordinates":[45.000000,7.000000],"type":"Point"},
"properties":{"Code postal":"8","Commune":"7","Coordonnee (X)":"45.0000000","Coordonnee (Y)":"7.0000000","Numero de maison":"7","Rue":"46"},"type":"Feature"},{"geometry":{"coordinates":[47.000000,7.000000],"type":"Point"},
"type":"Feature"}],"type":"FeatureCollection"}
```

Yes, I have my webpage (it's a map) who displays well but I don't see the different markers on the map because the post datas do not work.

Can you help me please ? Can you propose a solution ? 

Please help me.

Thank you

**Comment by champnic** [comment]
Hey @laurent0502 - Can you see if the `NavigateWithWebResourceRequest` function would work for you here?
https://docs.microsoft.com/en-us/microsoft-edge/webview2/reference/win32/icorewebview2_2?view=webview2-1.0.1108.44#navigatewithwebresourcerequest
